app/uk/gov/hmrc/tradergoodsprofilesdatastore/repositories/RecordsRepository.scala

```diff
@@ -152,4 +165,51 @@ class RecordsRepository @Inject() (
       case None => Future.successful(None)
     }
 
+  def filterRecords(eori: String, searchTerm: Option[String], field: Option[String]): Future[Seq[GoodsItemRecord]] =
+    searchTerm match {
+      case Some(value) =>
+        field match {
+          case Some("traderRef")        =>
```

Comment:
Can probably merge all of this into one - though we may want to validate that the fields are in our list of expected fields in the controller and return bad request if not

 collection
               .find[GoodsItemRecord](
                 Filters.and(
                   byEori(eori),
                   byField(field)
                 )
               )
               .sort(byLatest)
               .toFuture()

Reply:
Thanks for the suggestion. This has been updated.